go/vt/vtgate/executorcontext/vcursor_impl.go

```diff
@@ -805,7 +805,9 @@ func (vc *VCursorImpl) Execute(ctx context.Context, method string, query string,
 	}
 
 	qr, err := vc.executor.Execute(ctx, nil, method, session, vc.marginComments.Leading+query+vc.marginComments.Trailing, bindVars, false)
-	vc.setRollbackOnPartialExecIfRequired(err != nil, rollbackOnError)
+	// If there is no error, it indicates at least one successful execution,
+	// meaning a rollback should be triggered if a failure occurs later.
+	vc.setRollbackOnPartialExecIfRequired(err == nil, rollbackOnError)
```

Comment:
Is this the bug fix?

Reply:
This is another issue which I found, this was leading to calling `rollback savepoint` when not required as it should only be called when there is at least one successful execution.

Reply:
Wouldn't it be `!= nil` if there is one execution that passed but the other ones that failed?

Reply:
Either way, this need to be consistent between `ExecuteMultiShard` and `StreamExecuteMulti` and `Execute`

Reply:
Over there we check `len(errs) != len(rss)` this mean the errors and number of shards do not match which means that we have a `no error` from at least one shard.

Reply:
I am not sure I fully understand. To me it seems like we will almost always set the rollback command, even when there is no error. Leading to rollbacks even when we don't want them no?

Reply:
we have to save vtgate vttablet round trip calls as it adds to query latency. If there is no failure on lookup vindex DML (i.e. a success) then only we want to do a savepoint rollback if we encounter an error later.
we create savepoint as start of the incoming query execution and do savepoint rollback if we encounter a partial failure. So, we need to know if there was any success in between.